addons/payment_sips/models/sips.py

```diff
@@ -0,0 +1,237 @@
+    u'Č': u'C', u'č': u'c', u'Ć': u'C', u'ć': u'c', u'À': u'A', u'Á': u'A', u'Â': u'A', u'Ã': u'A',
+    u'Ä': u'A', u'Å': u'A', u'Æ': u'A', u'Ç': u'C', u'È': u'E', u'É': u'E', u'Ê': u'E', u'Ë': u'E',
+    u'Ì': u'I', u'Í': u'I', u'Î': u'I', u'Ï': u'I', u'Ñ': u'N', u'Ò': u'O', u'Ó': u'O', u'Ô': u'O',
+    u'Õ': u'O', u'Ö': u'O', u'Ø': u'O', u'Ù': u'U', u'Ú': u'U', u'Û': u'U', u'Ü': u'U', u'Ý': u'Y',
+    u'Þ': u'B', u'ß': u'Ss', u'à': u'a', u'á': u'a', u'â': u'a', u'ã': u'a', u'ä': u'a', u'å': u'a',
+    u'æ': u'a', u'ç': u'c', u'è': u'e', u'é': u'e', u'ê': u'e', u'ë': u'e', u'ì': u'i', u'í': u'i',
+    u'î': u'i', u'ï': u'i', u'ð': u'o', u'ñ': u'n', u'ò': u'o', u'ó': u'o', u'ô': u'o', u'õ': u'o',
+    u'ö': u'o', u'ø': u'o', u'ù': u'u', u'ú': u'u', u'û': u'u', u'ý': u'y', u'ý': u'y', u'þ': u'b',
+    u'ÿ': u'y', u'Ŕ': u'R', u'ŕ': u'r', u"`": u"'", u"´": u"'", u"„": u",", u"`": u"'",
+    u"´": u"'", u"“": u"\"", u"”": u"\"", u"´": u"'", u"&acirc;€™": u"'", u"{": u"",
+    u"~": u"", u"–": u"-", u"’": u"'", u">": u" ", u"<": u" "
+}
+
+
+def unormalize(text):
+    if not text:
+        return text
+
+    for inv_char, valid_char in UNORMALIZE_CHARS.items():
+        text = text.replace(inv_char, valid_char)
+    text = unicodedata.normalize('NFKD', text)
+    return text
+
+
+class AcquirerSips(models.Model):
+    _inherit = 'payment.acquirer'
+    # Fields
+    sips_merchant_id = fields.Char('API User Password',
+                                   required_if_provider='sips')
+    sips_secret = fields.Char('Secret', size=64, required_if_provider='sips')
+
+    # Methods
+    def _get_sips_urls(self, environment):
+        """ Worldline SIPS URLS """
+        url = {
+            'prod': 'https://payment-webinit.sips-atos.com/paymentInit',
+            'test': 'https://payment-webinit.simu.sips-atos.com/paymentInit', }
+
+        return {'sips_form_url': url.get(environment, url['test']), }
+
+    @api.model
+    def _get_providers(self):
+        providers = super(AcquirerSips, self)._get_providers()
+        providers.append(['sips', 'Sips'])
+        return providers
+
+    def _sips_generate_shasign(self, acquirer, values):
+        """ Generate the shasign for incoming or outgoing communications.
+
+        :param browse acquirer: the payment.acquirer browse record. It should
+                                have a shakey in shaky out
+        :param dict values: transaction values
+
+        :return string: shasign
+        """
+        assert acquirer.provider == 'sips'
+        data = values['Data']
+
+        key = u'002001000000001_KEY1'
+        if acquirer.environment == 'prod':
+            key = getattr(acquirer, 'sips_secret')
+
+        shasign = sha256(data + key)
+        return shasign.hexdigest()
+
+    @api.model
+    def sips_form_generate_values(self, partner_values, tx_values):
+        base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
+        acquirer = self.browse(id)
```

Comment:
id not defined (well it's the python `id()` method). Replace all `acquirer` by `self`